src/Workspaces/Remote/ServiceHub/Services/TodoComments/RemoteTodoCommentsIncrementalAnalyzer.cs

```diff
@@ -38,9 +39,10 @@ public override bool NeedsReanalysisOnOptionChanged(object sender, OptionChanged
 
         public override Task RemoveDocumentAsync(DocumentId documentId, CancellationToken cancellationToken)
         {
+            // Just report this back as there being no more comments for this document.
             return _endPoint.InvokeAsync(
-                nameof(ITodoCommentsListener.OnDocumentRemovedAsync),
-                new object[] { documentId },
+                nameof(ITodoCommentsListener.ReportTodoCommentDataAsync),
+                new object[] { documentId, Array.Empty<TodoCommentData>() },
                 cancellationToken);
```

Comment:
@uniqueiniquity Once this goes in, TS will be able to do the same.  To represent a document being deleted, just notify us that the set of comments for it went to 'empty'.  we'll clean things out in memory and will update the task list.